### PR TITLE
Makes reagent states subsystem a processing SS

### PR DESCRIPTION
**Pull request by yoyobatty**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs )

[why]: Copypasting subsystems are bad.

### CODE REVIEW

**Comment by Carbonhell** [comment]
This is not how the changelog nor the subsystems work, dude.

**Review by Carbonhell — CHANGES_REQUESTED** [review]
as i said on discord, subsystems do not use process. they use fire() at most, and even then i am sure you can avoid redefining it completely. Also if you dont want a changelog just do not write :cl: and /:cl:, you will break everything with the current changelog

File: hippiestation/code/controllers/subsystem/reagent_states.dm [hunk 1]
```diff
@@ -1,40 +1,16 @@
...
-			return
-
-/datum/controller/subsystem/reagent_states/Recover()
-	processing = SSreagent_states.processing
\ No newline at end of file
+/datum/controller/subsystem/processing/reagent_states/fire(resumed = FALSE)
+	. = ..()
+	if(!deleting && cost > 2000)
+		deleting = TRUE
+		for(var/I in GLOB.smoke)
```
Comment (Carbonhell) [inline hippiestation/code/controllers/subsystem/reagent_states.dm]: This stuff could be locking the server up if those lists are big. Not sure how big they can be but a MC_CHECK_TICK check at the end of each for cycle could help.

**Review by Carbonhell — CHANGES_REQUESTED** [review]

**Comment by Carbonhell** [comment]
DNM this unless you know exactly what you are doing, also dnm right now since it is not ready

**Comment by yoyobatty** [comment]
@Carbonhell ready

File: hippiestation/code/controllers/subsystem/reagent_states.dm [hunk 2]
```diff
@@ -1,40 +1,18 @@
...
-				deleting = FALSE
-			return
-
-/datum/controller/subsystem/reagent_states/Recover()
-	processing = SSreagent_states.processing
\ No newline at end of file
+/datum/controller/subsystem/processing/reagent_states/fire(resumed = FALSE)
+	. = ..()
+	if(MC_TICK_CHECK)//if subsystems pause
+		if(!deleting && cost > 1000)
```
Comment (Carbonhell) [inline hippiestation/code/controllers/subsystem/reagent_states.dm:11]: why did you lower the cost

Comment (yoyobatty) [reply hippiestation/code/controllers/subsystem/reagent_states.dm:11]: Because anything over 1000 is already ridiculous. Let's keep this number for now.

**Review by Carbonhell — APPROVED** [review]

**Comment by Carbonhell** [comment]
Please be ready to fix anything if this breaks.

**Pull request merged**